internal/operands/common-templates/reconcile_test.go

```diff
@@ -352,39 +320,132 @@ var _ = Describe("Common-Templates operand", func() {
+
+		BeforeEach(func() {
+			request.Instance.Spec.EnableMultipleArchitectures = ptr.To(true)
+			request.Instance.Spec.Cluster = &ssp.Cluster{
+				WorkloadArchitectures:     []string{amd64, arm64},
+				ControlPlaneArchitectures: []string{amd64},
+			}
+
+			armTemplates = []templatev1.Template{}
+			for _, template := range testTemplates {
+				if GetTemplateArch(&template) == arm64 {
+					armTemplates = append(armTemplates, template)
+				}
+			}
+		})
+
+		It("should create common-template resources", func() {
+			_, err := operand.Reconcile(&request)
+			Expect(err).ToNot(HaveOccurred())
+			for _, template := range testTemplates {
+				template.Namespace = namespace
+				ExpectResourceExists(&template, request)
+			}
+		})
+
+		It("should reconcile predefined labels", func() {
+			const (
+				defaultOsLabel = "template.kubevirt.io/default-os-variant"
+				testLabel      = "some.test.label"
+			)
+
+			for i := range testTemplates {
+				template := testTemplates[i].DeepCopy()
+				template.Namespace = namespace
+				template.Labels[defaultOsLabel] = "true"
+				template.Labels[testLabel] = "test"
+				Expect(request.Client.Create(request.Context, template)).To(Succeed())
+			}
+
+			_, err := operand.Reconcile(&request)
+			Expect(err).ToNot(HaveOccurred())
+
+			for i := range testTemplates {
+				key := client.ObjectKey{
+					Name:      testTemplates[i].Name,
+					Namespace: namespace,
+				}
+				template := &templatev1.Template{}
+				Expect(request.Client.Get(request.Context, key, template)).To(Succeed())
+
+				Expect(template.Labels).ToNot(HaveKey(defaultOsLabel))
+				Expect(template.Labels).To(HaveKey(testLabel))
+			}
+		})
+
+		It("should remove templates if architecture is no longer available", func() {
+			_, err := operand.Reconcile(&request)
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, template := range armTemplates {
+				template.Namespace = namespace
+				ExpectResourceExists(&template, request)
+			}
+
+			request.Instance.Spec.Cluster.WorkloadArchitectures = []string{amd64}
+
+			_, err = operand.Reconcile(&request)
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, template := range armTemplates {
+				template.Namespace = namespace
+				ExpectResourceNotExists(&template, request)
+			}
+		})
+
+		It("should remove templates multi-architecture is disabled", func() {
+			_, err := operand.Reconcile(&request)
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, template := range armTemplates {
+				template.Namespace = namespace
+				ExpectResourceExists(&template, request)
+			}
+
+			request.Instance.Spec.EnableMultipleArchitectures = nil
+
+			_, err = operand.Reconcile(&request)
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, template := range armTemplates {
+				template.Namespace = namespace
+				ExpectResourceNotExists(&template, request)
+			}
+		})
+	})
 })
 
 func getTestTemplates() []templatev1.Template {
-	return []templatev1.Template{{
+	const amdArch = "amd64"
+	const armArch = "arm64"
```

Comment:
Make these top level consts in the file?

Reply:
done